include/private/common/libevent_reactor.hpp

```diff
@@ -195,62 +151,32 @@ class LibeventReactor : public Reactor, public NonCopyable, public NonMovable {
         }
     }
 
-    void pollin_once(socket_t fd, double timeo, Callback<Error> &&cb) override {
-        pollfd(fd, MK_POLLIN, timeo, [cb = std::move(cb)](Error err, short) {
-            cb(std::move(err));
-        });
+    static void pollfd_cb(short evflags, void *opaque) {
+        auto cbp = static_cast<mk::Callback<mk::Error, short> *>(opaque);
+        mk::Error err = mk::NoError();
+        assert((evflags & (~(EV_TIMEOUT | EV_READ | EV_WRITE))) == 0);
+        if ((evflags & EV_TIMEOUT) != 0) {
+            err = mk::TimeoutError();
+        }
+        // In case of exception here, the stack is going to unwind, tearing down
+        // the libevent loop and leaking forever `cbp` and the event once that
+        // was used to invoke this callback.
+        (*cbp)(std::move(err), evflags);
+        delete cbp;
     }
 
-    void pollout_once(
-            socket_t fd, double timeo, Callback<Error> &&cb) override {
-        pollfd(fd, MK_POLLOUT, timeo, [cb = std::move(cb)](Error err, short) {
-            cb(std::move(err));
-        });
-    }
+  private:
+    // ## Private attributes
+
+    event_base *evbase = nullptr;
+    Worker worker;
 };
 
 } // namespace mk
 
-/*-
-      ___   _ _      _                           _ _ _             _
-     / __| | (_)_ _ | |____ _ __ _ ___   __ __ _| | | |__  __ _ __| |__ ___
-    | (__  | | | ' \| / / _` / _` / -_) / _/ _` | | | '_ \/ _` / _| / /(_-<
-     \___| |_|_|_||_|_\_\__,_\__, \___| \__\__,_|_|_|_.__/\__,_\__|_\_\/__/
-                             |___/
-
-
-    Here we have all the callbacks called directly by C code.
-*/
-
-static inline void mk_call_later_cb(
-        evutil_socket_t, short evflags, void *opaque) {
-    assert((evflags & (~(EV_TIMEOUT))) == 0);
-    auto cbp = static_cast<mk::Callback<> *>(opaque);
-    // In case of exception here, the stack is going to unwind, tearing down
-    // the libevent loop and leaking forever `cbp` and the event once that was
-    // used to invoke this callback.
-    (*cbp)();
-    delete cbp;
-}
```

Comment:
Callback no longer needed since now we have unified handling delayed calls and polling sockets.